File: GameClient/Assets/Scripts/GameLogic/State/LoginState.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class LoginState : StateBase
{
    public LoginState()
    {
        currentState = StateType.Login;
    }

    public override void Enter()
    {
        Debug.Log("enter loginstate");        
        LoginController.Instance.ShowView();
        LoginController.Instance.RegisterProto();
    }

    public override void Excute()
    {
        
    }

    public override void Exit()
    {
        Debug.Log("exit loginstate");        
        LoginController.Instance.CloseView();
        LoginController.Instance.UnregisterProto();
    }

}

File: GameClient/Assets/HotFix/Lua/FSM/LoginState.lua
local  loginState = class(require "FSM/BaseState")

function loginState:enter( )
	global.loginController:showView()	
	global.loginController:registerProto()
end

function loginState:exit(  )
	global.loginController:closeView()
	global.loginController:unregisterProto()
end

return loginState

